src/QsCompiler/SyntaxProcessor/TypeInference/InferenceContext.fs

```diff
@@ -323,7 +325,20 @@ type InferenceContext(symbolTracker: SymbolTracker) =
         |> Seq.map (fun item -> diagnostic item.Key item.Value)
         |> Seq.toList
 
-    member context.UseStatementPosition position = statementPosition <- position
+    member context.UseStatementPosition position =
```

Comment:
It seems like `UseStatementPosition` is not really used anymore, is it kept here just because it is a public member and would break someone trying to use it? If that's the case, should it be deprecated?

Reply:
I debated whether to deprecate it, but given that the inference context is part of the public API, it could be used outside the Q# compiler, and then it would likely be more convenient to work with UseStatementPosition only rather than to split that into absolute and relative location. I hence decided against deprecating it, but I could go either way.

Reply:
Makes sense to not deprecate it for convenience outside the Q# compiler.